StreamChatClient_v3/Database/DTOs/ChannelDTO_Tests.swift

```diff
@@ -17,6 +17,109 @@ class ChannelModelDTO_Tests: XCTestCase {
     func test_channelPayload_isStoredAndLoadedFromDB() {
         let channelId = UUID().uuidString
         
+        let payload = dummyPayload(with: channelId)
+        
+        // Asynchronously save the payload to the db
+        database.write { session in
+            session.saveChannel(payload: payload)
+        }
+        
+        // Load the channel from the db and check the fields are correct
+        var loadedChannel: ChannelModel<DefaultDataTypes>? {
+            database.viewContext.loadChannel(id: channelId)
+        }
+        
+        AssertAsync {
+            // Channel details
+            Assert.willBeEqual(ChannelId(type: ChannelType(rawValue: payload.channel.typeRawValue), id: channelId),
+                               loadedChannel?.id)
+            
+            Assert.willBeEqual(payload.channel.extraData, loadedChannel?.extraData)
+            Assert.willBeEqual(payload.channel.typeRawValue, loadedChannel?.type.rawValue)
+            Assert.willBeEqual(payload.channel.lastMessageDate, loadedChannel?.lastMessageDate)
+            Assert.willBeEqual(payload.channel.created, loadedChannel?.created)
+            Assert.willBeEqual(payload.channel.updated, loadedChannel?.updated)
+            Assert.willBeEqual(payload.channel.deleted, loadedChannel?.deleted)
+            
+            // Config
+            Assert.willBeEqual(payload.channel.config.reactionsEnabled, loadedChannel?.config.reactionsEnabled)
+            Assert.willBeEqual(payload.channel.config.typingEventsEnabled, loadedChannel?.config.typingEventsEnabled)
+            Assert.willBeEqual(payload.channel.config.readEventsEnabled, loadedChannel?.config.readEventsEnabled)
+            Assert.willBeEqual(payload.channel.config.connectEventsEnabled, loadedChannel?.config.connectEventsEnabled)
```

Comment:
Why you can't compare channels themselves?
```
Assert.willBeEqual(payload.channel, loadedChannel)
```

Reply:
This is a great question! I thought about it and I think it's more intuitive to implement a custom `==` for model objects just based on their `id`. Example:
```
let user1 = User(id: "alex")
user1.isOnline = true
let user2 = User(id: "alex")
user1.isOnline = false

user1 == user2 // ???
```
What would you expect the result of the above would be? I'd expect that the result would be `true`.

It can also cause problems when you put the object to a set:
```
let user1 = User(id: "alex")
user1.isOnline = true
let user2 = User(id: "alex")
user1.isOnline = false

channel.members: Set<User> = [user1, user2]
// Here you want to have just one user in the set
```

Reply:
@b-onc it would be great to have your point on this, too

Reply:
@VojtaStavik I think that's a good point you made, and I can't come up with a reason why I'd like to differentiate between users with same id. If we do that, we'll find ourselves doing `user1.id == user2.id` very often, but then again comparing id's is more explicit and would be easier to understand, in a programmer's point of view. If I hadn't seen this discussion and used `==`, I'd expect the objects to be exactly the same. So, overriding default equality operator for our use case feels like hiding the intention a bit.

Reply:
I think the biggest issue is with the `Set` example. For me, that was the biggest reason for using only `id` for `hash` and for `==`.

Reply:
I'd say that justifies the decision 👍

Reply:
We're already using simpler comparison:
```swift
    // for User
    public static func == (lhs: User, rhs: User) -> Bool {
        lhs.id == rhs.id
    }

    // for Channel
    public static func == (lhs: Channel, rhs: Channel) -> Bool {
        lhs.cid == rhs.cid
    }
```